Packages/com.unity.inputsystem/InputSystem/Plugins/Switch/NPad.cs

```diff
@@ -118,174 +110,351 @@ public NPadInputState WithButton(Button button, bool value = true)
+        public FourCC GetTypeStatic() { return Type; }
+        public static NpadDeviceIOCTLStartSixAxisSensor Create()
         {
-            FullKey = (1 << 0),
-            Handheld = (1 << 1),
-            JoyDual = (1 << 2),
-            JoyLeft = (1 << 3),
-            JoyRight = (1 << 4),
+            return new NpadDeviceIOCTLStartSixAxisSensor
+            {
+                baseCommand = new InputDeviceCommand(Type, kSize),
+            };
         }
+    }
+
+    [StructLayout(LayoutKind.Explicit, Size = kSize)]
+    public struct NpadDeviceIOCTLStopSixAxisSensor : IInputDeviceCommandInfo
+    {
+        public static FourCC Type { get { return new FourCC("SXSP"); } }
+        public const int kSize = InputDeviceCommand.kBaseCommandSize;
+
+        [FieldOffset(0)]
+        public InputDeviceCommand baseCommand;
 
-        public long ShowControllerSupportUI()
+        public FourCC GetTypeStatic() { return Type; }
+        public static NpadDeviceIOCTLStopSixAxisSensor Create()
         {
-            var command = NPadShowControllerSupportUICommand.Create();
+            return new NpadDeviceIOCTLStopSixAxisSensor
+            {
+                baseCommand = new InputDeviceCommand(Type, kSize),
+            };
+        }
+    }
+}
 
-            return ExecuteCommand(ref command);
+namespace UnityEngine.Experimental.Input.Plugins.Switch
+{
+	/// <summary>
+	/// An NPad controller for Switch, which can be a Joy-Con.
+	/// </summary>
+	/// <seealso cref="NPadInputState"/>
+	[InputControlLayout(stateType = typeof(NPadInputState))]
+	public class NPad : Gamepad
+	{
+		public ButtonControl leftSL { get; private set; }
+		public ButtonControl leftSR { get; private set; }
+		public ButtonControl rightSL { get; private set; }
+		public ButtonControl rightSR { get; private set; }
+
+        public Vector3Control acceleration { get; private set; }
+        public QuaternionControl attitude { get; private set; }
+        public Vector3Control angularVelocity { get; private set; }
+
+        public enum Orientation
+		{
+			Vertical,
+			Horizontal,
+			Default = Vertical,
+		}
+
+		public enum NpadId : int
+		{
+			No1 = 0x00,
+			No2 = 0x01,
+			No3 = 0x02,
+			No4 = 0x03,
+			No5 = 0x04,
+			No6 = 0x05,
+			No7 = 0x06,
+			No8 = 0x07,
+			Handheld = 0x20,
+			Debug = 0xFF,
+		}
+
+		[Flags]
+		public enum NpadStyle
+		{
+			FullKey = (1 << 0),
+			Handheld = (1 << 1),
+			JoyDual = (1 << 2),
+			JoyLeft = (1 << 3),
+			JoyRight = (1 << 4),
+		}
+
+		public struct JoyConColor
+		{
+			public Color32 Main;
+			public Color32 Sub;
+		}
+
+		public Orientation orientation
+		{
+			get
+			{
+                RefreshConfigurationIfNeeded();
+				return m_Orientation;
+			}
+		}
+		public int controllerID
+		{
+			get
+			{
+                RefreshConfigurationIfNeeded();
+				return m_ControllerID;
+			}
+		}
+		public NpadId npadID
+		{
+			get
+			{
+                RefreshConfigurationIfNeeded();
+				return m_NpadId;
+			}
+		}
```

Comment:
What's the difference between npadID and controllerID? Preferably add XML doc comments. Maybe there's a better name for controllerID?

Reply:
I imagined they can be different. NPad ID represents a port where a controller is connected to. For example, you can be a 1st player even if you use JoyCons which is connect as NPad ID 2, but it doesn't do anything for now. I will remove it.

Reply:
That sounds useful. Maybe call it `portNumber`?

Also wondering, is there a relation between `userId` (coming from `InputDevice`) and `controllerID`? Might be worth looking at the Xbox One code for parallels.

Naming convention wise, think we should use `Id` instead of `ID` everywhere.